minorminer/layout/construction.py

```diff
@@ -0,0 +1,141 @@
+
+    Parameters
+    ----------
+    T : NetworkX graph
+        The graph you are embedding into (target).
+    placement : dict
+        A mapping from vertices of S (keys) to vertices of T (values).
+    second : bool
+        If True, gets the closed 2nd neighborhood of each vertex. If False, get the closed 1st neighborhood of each
+        vertex. 
+
+    Returns
+    -------
+    chains: dict
+        A mapping from vertices of S (keys) to chains of T (values).
+    """
+    return {u: closed_neighbors(T, v, second=second) for u, v in placement.items()}
+
+
+def extend(S, T, placement, nhbd_ext=False):
+    """
+    Given a placement (a map phi from vertices of S to vertices of T), form the chain that is the union of phi(u) and 
+    "optimal" open paths between phi(u) and each vertex in phi(N_S(u)). Optimal paths are shortest nonoverlapping paths.
+
+    Parameters
+    ----------
+    S : NetworkX graph
+        The graph you are embedding (source).
+    T : NetworkX graph
+        The graph you are embedding into (target).
+    placement : dict
+        A mapping from vertices of S (keys) to vertices of T (values).
+    nhbd_ext : bool (default False)
+        If True, return the closed neighborhood of each chain extension. If False, just return the chain extensions.
+
+    Returns
+    -------
+    chains: dict
+        A mapping from vertices of S (keys) to chains of T (values).
+    """
+    extended_chains = extend_chains(S, T, placement)
+
+    if not nhbd_ext:
+        return extended_chains
+
+    for u, C in extended_chains.items():
+        for v in list(C):
+            extended_chains[u].update(closed_neighbors(T, v))
+
+    return extended_chains
+
+
+def extend_chains(S, T, placement):
+    """
+    Extend initial_chains in T so that their structure matches that of S. That is, form an overlap embedding of S in T
+    where the initial_chains are subsets of the overlap embedding chains. This is done via minorminer.
+
+    Parameters
+    ----------
+    S : NetworkX graph
+        The graph you are embedding (source).
+    T : NetworkX graph
+        The graph you are embedding into (target).
+    placement : dict
+        A mapping from vertices of S (keys) to vertices of T (values).
+
+    Returns
+    -------
+    extended_chains: dict
+        A mapping from vertices of S (keys) to chains of T (values).
+    """
+    # Extend the chains to minimal overlapped embedding
+    initial_chains = {u: [v] for u, v in placement.items()}
+    miner = mm.miner(S, T, initial_chains=initial_chains)
+    extended_chains = defaultdict(set)
+    for u in S:
+        # Revert to the initial_chains and compute the embedding where you tear-up u.
+        emb = miner.quickpass(
+            [u], clear_first=True, overlap_bound=S.number_of_nodes()
+        )
+
+        # Add the new chain for u and the singleton one too (in case it got moved in emb)
+        extended_chains[u].update(set(emb[u]).union(initial_chains[u]))
```

Comment:
this looks... suspicious.  specifically, `emb[u]` might not be connected to `initial_chains[u]`... is this a good idea?

Reply:
Now that you mention it, this may not be doing what I thought it was doing. The intention was to use the `placement` as fixed points and run an extension from each fixed point. 

In light of this I'll remove this strategy, and if we want, I can bring a more proper one into the fold with the more current codebase.

Reply:
Sounds like a good plan.